fcrepo-http-api/src/main/java/org/fcrepo/http/api/services/HttpRdfService.java

```diff
@@ -0,0 +1,142 @@
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.fcrepo.http.api.services;
+
+import static org.apache.jena.rdf.model.ModelFactory.createDefaultModel;
+import static org.fcrepo.kernel.api.rdf.DefaultRdfStream.fromModel;
+import static org.apache.jena.riot.RDFLanguages.contentTypeToLang;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import java.io.InputStream;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.core.MediaType;
+import org.apache.jena.atlas.RuntimeIOException;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RiotException;
+import org.fcrepo.http.commons.api.rdf.HttpIdentifierConverter;
+import org.fcrepo.kernel.api.exception.MalformedRdfException;
+import org.fcrepo.kernel.api.exception.RepositoryRuntimeException;
+import org.fcrepo.kernel.api.RdfStream;
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.slf4j.Logger;
+
+/**
+ * A service that will translate the resourceURI to fedora ID in the Rdf InputStream
+ *
+ * @author bseeger
+ * @author bbpennel
+ * @since 2019-11-07
+ */
+
+public class HttpRdfService {
+
+    private static final Logger LOGGER = getLogger(HttpRdfService.class);
+
+    /**
+     * Parse the request body to a Model, with the URI to Fedora ID translations done.
+     *
+     * @param resource the fedora resource
+     * @param stream the input stream containing the RDF
+     * @param contentType the media type of the RDF
+     * @param idTranslator the uri to fedora resource ID translator
+     * @return RdfStream containing triples from request body, with fedora IDs in them
+     * @throws MalformedRdfException in case rdf json cannot be parsed
+     * @throws BadRequestException in the case where the RDF syntax is bad
+     */
+    public Model bodyToInternalModel(final FedoraResource resource, final InputStream stream,
+                                     final MediaType contentType, final HttpIdentifierConverter idTranslator)
+                                     throws RepositoryRuntimeException, BadRequestException {
+
+        final Model model = parseBodyAsModel(stream, contentType, resource);
+
+        final StmtIterator stmtIterator = model.listStatements();
```

Comment:
I was testing this some, and I'm finding that it's not behaving the way I remember it working before. It appears to iterate through the newly added items in the main look. So I think we're going to have to add the new statements to a list and insert them after the loop like you suggested yesterday. 

There were a couple other issues (only need to add/remove when there is a change, the object might not be a literal, etc) and I ended up finding I had to update it to figure out what feedback to give, so sorry if this is a bit much but here is a suggested updated implementation:
```
        final Model model = parseBodyAsModel(stream, contentType, resource);

        final var insertStatements = new ArrayList<Statement>();
        final StmtIterator stmtIterator = model.listStatements();
        while (stmtIterator.hasNext()) {
            final Statement stmt = stmtIterator.nextStatement();

            final String originalSubj = stmt.getSubject().getURI();
            final String subj = idTranslator.convert(originalSubj);

            var obj = stmt.getObject();
            if (stmt.getObject().isResource()) {
                final String objString = stmt.getObject().asResource().getURI();
                obj = model.getResource(idTranslator.convert(objString));
            }

            if (!subj.equals(originalSubj) || !obj.equals(stmt.getObject())) {
                insertStatements.add(new StatementImpl(model.getResource(subj),
                        stmt.getPredicate(),
                        obj));

                stmtIterator.remove();
            }
        }

        model.add(insertStatements);

        LOGGER.debug("Model: {}", model);
        return model;
```

Reply:
Thanks, Ben.  Putting this in.